backend/test/api/app_test.go

```
@@ -26,7 +27,7 @@ func TestListApp(t *testing.T) {
 		require.NotNil(t, appsDB)
 
 		// fetch apps from the API
-		url := fmt.Sprintf("%s/api/apps", testServerURL)
+		url := fmt.Sprintf("%s/api/apps", os.Getenv("NEBRASKA_TEST_SERVER_URL"))
```

Instead of using `os.Getenv` create a function that returns the default value if the value is not found in env. I see that the value is being set in the `TestMain` function but when we want to run a single test it will fail as the value is not set manually. 
Running a single test like `go test -run TestListApp  .` this will fail.

---

Thanks, good idea.

---

I tried this and TestListApp works...

```
cd backend && make test-service-down test-service-up && cd test/api && NEBRASKA_RUN_SERVER_TESTS=1 NEBRASKA_TEST_SERVER_URL="http://localhost:8002" NEBRASKA_DB_URL="postgres://postgres:nebraska@127.0.0.1:8001/nebraska_tests?sslmode=disable&connect_timeout=10" go test -run TestListApp
```

---

Thanks @yolossn 

The issue was when running a single test, but NEBRASKA_TEST_SERVER_URL is not set (so the default will be used).

This is what I found:
- There was an incorrect port (pushed a fix just now for this)
- TestMain runs even if you select one of the tests to run.

This is the command that should work (note the missing NEBRASKA_TEST_SERVER_URL):
```
cd backend && make test-service-down test-service-up && cd test/api && NEBRASKA_RUN_SERVER_TESTS=1 NEBRASKA_DB_URL="postgres://postgres:nebraska@127.0.0.1:8001/nebraska_tests?sslmode=disable&connect_timeout=10" go test -run TestListApp
```